### PR TITLE
Clarify the documented behavior and lifetime of key handles

### PR DESCRIPTION
**Pull request by athoelke**
This accompanies changes in the overview documentation relating to the lifetime and behavior of key handles.

### CODE REVIEW

**Comment by Patater** [comment]
CI is probably OK on these changes. To get a good CI run, please rebase this PR on the latest `development `branch.

File: include/psa/crypto.h [hunk 1]
```diff
@@ -394,15 +400,19 @@ psa_status_t psa_open_key(psa_key_id_t id,
...
- * not affected and can be opened again later with psa_open_key().
+ * If this is the last open handle to a persistent key, then closing the handle
+ * will free all resources associated with the key in volatile memory. The key
+ * data in persistent storage is not affected and can be opened again later
+ * with a call to psa_open_key().
  *
- * If the key is currently in use in a multipart operation,
- * the multipart operation is aborted.
+ * Closing the key handle makes the handle invalid, and the key handle
+ * must not be used again by the application..
```
Comment (adrianlshaw) [inline include/psa/crypto.h]: Single full stop needed.

File: include/psa/crypto.h [hunk 2]
```diff
@@ -394,15 +400,19 @@ psa_status_t psa_open_key(psa_key_id_t id,
...
+ *
+ * If the key is currently in use in a multipart operation, then closing the
+ * last handle to the key will abort the multipart operation.
```
Comment (adrianlshaw) [inline include/psa/crypto.h]: I slightly prefer "last remaining" in order to (attempt to) disambiguate "last"

**Comment by athoelke** [comment]
> CI is probably OK on these changes. To get a good CI run, please rebase this PR on the latest `development `branch.

Hi @Patater - should I create a separate branch and PR against `development` for the CI run, and leave this one here for discussion and merging? (presuming that psa-api-1.0-beta is the right target?)

**Comment by Patater** [comment]
Please target the `development` branch. We've recently merged `psa-api-1.0-beta` into `development` and so `psa-api-1.0-beta` has served its purpose and isn't used for future development work. We probably need to have `psa-crypto` repo point at the development branch now when it generates documents.

**Comment by athoelke** [comment]
New PR raised #235 rebased and targeting `development`.

**Comment by Patater** [comment]
Closing in favor of #235

**Pull request closed without merging**